### PR TITLE
Add support for EXISTS and NOT EXISTS

### PR DESCRIPTION
**Pull request by ryanhossain9797**
SELECT foo FROM TableA `WHERE EXISTS`(
  SELECT * FROM TableB WHERE(TableA.foo = TableB.bar)
)

SELECT foo FROM TableA `WHERE NOT EXISTS`(
  SELECT * FROM TableB WHERE(TableA.foo = TableB.bar)
)

### CODE REVIEW

**Comment by panarch** [comment]
Wow, you even found issues my yourself and did it, thanks so so much!
It looks very good but for now I have to leave now so I'll review this little bit later, in a single day.
Thanks!

**Review by panarch — CHANGES_REQUESTED** [review]

File: src/executor/filter.rs [hunk 1]
```diff
@@ -165,6 +165,7 @@ fn check_expr<'a, T: 'static + Debug>(
 
             Ok(negated ^ (evaluate(low)? <= target && target <= evaluate(high)?))
         }
+        Expr::Exists(expr) => Ok(select(storage, expr, filter_context)?.next().is_some()),
```
[the remark this answers is not in the document]
Comment (ryanhossain9797) [reply src/executor/filter.rs]: My bad, didn't notice it was different type. Fixed

File: src/tests/filter.rs [hunk 2]
```diff
@@ -59,3 +59,59 @@ CREATE TABLE Test (
     );
     assert_eq!(expected, found);
 }
+
```
[the remark this answers is not in the document]
Comment (ryanhossain9797) [reply src/tests/filter.rs]: Made these changes too

**Review by panarch — APPROVED** [review]
Wow, you even did work `tests/filter.rs` codes, thanks so much.
All looks beautiful, merging 👍

**Pull request merged**